pkg/storaged/client.js

```diff
@@ -443,24 +443,22 @@
 
     function enable_udisks_features() {
         if (!client.manager.valid)
-            return cockpit.resolve();
+            return Promise.resolve();
         if (!client.manager.EnableModules)
-            return cockpit.resolve();
+            return Promise.resolve();
         return client.manager.EnableModules(true).then(
             function() {
-                const defer = cockpit.defer();
                 client.manager_lvm2 = proxy("Manager.LVM2", "Manager");
                 client.manager_iscsi = proxy("Manager.ISCSI.Initiator", "Manager");
-                Promise.allSettled([client.manager_lvm2.wait(), client.manager_iscsi.wait()]).then(() => {
-                    client.features.lvm2 = client.manager_lvm2.valid;
-                    client.features.iscsi = (client.manager_iscsi.valid &&
-                                                      client.manager_iscsi.SessionsSupported !== false);
-                    defer.resolve();
-                });
-                return defer.promise;
+                return Promise.allSettled([client.manager_lvm2.wait(), client.manager_iscsi.wait()])
+                        .then(() => {
+                            client.features.lvm2 = client.manager_lvm2.valid;
+                            client.features.iscsi = (client.manager_iscsi.valid &&
+                                                            client.manager_iscsi.SessionsSupported !== false);
+                        });
             }, function(error) {
                 console.warn("Can't enable storaged modules", error.toString());
-                return cockpit.resolve();
+                return Promise.resolve();
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)